modules/javafx.base/src/main/java/com/sun/javafx/PreviewFeature.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javafx;
+
+import com.sun.javafx.runtime.VersionInfo;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Using a preview feature requires an opt-in from application developers by specifying the
+ * {@code javafx.enablePreview=true} system property. This class verifies that the application
+ * has opted into preview features.
+ */
+public enum PreviewFeature {
```

Comment:
Why enum?
Maybe use a regular class with a public constructor should do it.
What do you think?

Reply:
The idea is to make it easy and obvious to use from preview code:
```java
PreviewFeature.MY_PREVIEW_FEATURE.checkEnabled();
```
In addition to that, any IDE can easily list all uses of an enum constant, which gives you an overview where APIs of a particular feature are located. After all, enums are just regular classes with singleton instances.

By the way, preview features can't be individually enabled or disabled. The point of checking on a enum constant is to generate an appropriate error message (including the name of the feature) in case the user didn't enable preview features.

Reply:
my only objection is that enum implies the use of a switch pattern, which is not really applicable here.